deepautoencoder/stacked_autoencoder.py

```diff
@@ -127,10 +140,47 @@ def run(self, data_x, data_x_, hidden_dim, activation, loss, lr,
                 print('epoch {0}: global loss = {1}'.format(i, l))
         # debug
         # print('Decoded', sess.run(decoded, feed_dict={x: self.data_x_})[0])
-        self.weights.append(sess.run(encode['weights']))
-        self.biases.append(sess.run(encode['biases']))
+
+        self.weights.append(encode['weights'])
+        self.biases.append(encode['biases'])
         return sess.run(encoded, feed_dict={x: data_x_})
 
+    def finetune(self, data_x, data_x_, loss, lr, print_step, epoch,
```

Comment:
Fine tuning is a supervised process, isn't it? I mean that you take in the labels (train_Y from test.py) and perform a softmax classification in the last layer of the autoencoder and update (or fine-tune) the weights of the whole autoencoder ! :smiley: